src/Console/Command/ReleaseCommand.php

```diff
@@ -127,7 +128,12 @@ protected function findLastVersion(SymfonyStyle $style): void
         $style->text('The previous version on this branch is <info>' . $currentVersion . '</info>.');
 
         $style->text('A new release will introduce the following changes:');
-        $style->listing($this->changelogGenerator->getUnreleasedChangelog()->getUnreleasedChanges());
+
+        $filteredChangeLogElements = $this->filterChangeLogElements(
+            $this->changelogGenerator->getUnreleasedChangelog()->getUnreleasedChanges()
```

Comment:
Filtering the merge requests is not a responsibility of this class.

I feel this logic should be included inside the change log generator class that is being used.
That will also remove the need for the extra check in the markdownFormatter since that will only format anything that is inside the Changelog which is being returned from the ChangeLogGenerator.

Maybe even better would be if this was an optional configuration for our changelog generator, but lets not include that in the scope for now.

Reply:
You mean this file: `PullRequestChangelogGenerator.php`

Reply:
Its optional to use; Not an optional configuration.
As this is an open-source project, some users might have a workflow that includes merges. Or they would want to use a different tag to specify merge requests with.

Maybe this issue is only specific with our workflow and others don't even need it. Maybe they are doing trunk-based development?

Those kind of questions mean that this feature requires a little more thought; Now, or later.